iree/compiler/Conversion/Common/Transforms.cpp

```diff
@@ -317,10 +317,9 @@ LogicalResult defineWorkgroupCountRegion(
     OpBuilder &builder, FuncOp funcOp,
     WorkgroupCountRegionBuilder regionBuilder) {
   IREE::HAL::ExecutableEntryPointOp entryPointOp = getEntryPoint(funcOp);
-  if (!entryPointOp)
+  if (!entryPointOp) {
     return funcOp.emitOpError("unable to find corresponding entry point op");
-  if (entryPointOp.getBody())
```

Comment:
Is this due to we put multiple linalg ops in the same dispatch region and the `TileAndVectorizeInOneWorkgroupPass` is trying to update for both of them? At least that's what I saw when this error was hit. Does it mean we are missing a proper operand-result tie in `DispatchLinalgOnTensorsPass`?

Reply:
Actually this is a good catch. THis is the repro I had

```
  hal.interface.binding @ro0, set=0, binding=0, type="StorageBuffer", access="Read"
    hal.interface.binding @ro1, set=0, binding=1, type="StorageBuffer", access="Read"
    hal.interface.binding @wo2, set=0, binding=2, type="StorageBuffer", access="Write|Discard"
  }
  hal.executable.target @vulkan, filter="vulkan*" {
    hal.executable.entry_point @serving_default_dispatch_21 attributes {
      interface = @legacy_io, ordinal = 0 : i32,
      signature = (!flow.dispatch.tensor<readonly:1x4x384xf32>, !flow.dispatch.tensor<readonly:1x4x384x384xf32>,
        !flow.dispatch.tensor<writeonly:1x4x384x384xf32>) -> ()}
    module attributes {spv.target_env = #spv.target_env<#spv.vce<v1.3, [Shader, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative], [SPV_KHR_storage_buffer_storage_class]>, SwiftShader:CPU\
, {cooperative_matrix_properties_nv = [], max_compute_shared_memory_size = 16384 : i32, max_compute_workgroup_invocations = 128 : i32, max_compute_workgroup_size = dense<[128, 128, 64]> : vector<3xi32>, subgroup_size = 4 : i32}>}  {
      func @serving_default_dispatch_21() {
        %c0 = constant 0 : index
        %cst = constant 0x7FC00000 : f32
        %0 = hal.interface.binding.subspan @legacy_io::@ro1[%c0] : memref<1x4x384xf32>
        %1 = hal.interface.binding.subspan @legacy_io::@wo2[%c0] : memref<1x4x384xf32>
        %2 = hal.interface.binding.subspan @legacy_io::@ro0[%c0] : memref<1x4x384x384xf32>
        linalg.copy(%0, %1) : memref<1x4x384xf32>, memref<1x4x384xf32> 
        linalg.generic {indexing_maps = [affine_map<(d0, d1, d2) -> (0, d0, d1, d2)>, affine_map<(d0, d1, d2) -> (0, d0, d1)>], iterator_types = ["parallel", "parallel", "reduction"]} ins(%2 : memref<1x4x384x384xf32>) outs(%1 : memref<1x4x384xf32>) {
        ^bb0(%arg0: f32, %arg1: f32):  // no predecessors
          %3 = cmpf ogt, %arg0, %arg1 : f32
          %4 = select %3, %arg0, %arg1 : f32
          %5 = cmpf uno, %arg0, %arg1 : f32
          %6 = select %5, %cst, %4 : f32
          linalg.yield %6 : f32
        }
        return
      }
      hal.interface @legacy_io attributes {sym_visibility = "private"} {
        hal.interface.binding @ro0, set=0, binding=0, type="StorageBuffer", access="Read"
        hal.interface.binding @ro1, set=0, binding=1, type="StorageBuffer", access="Read"
        hal.interface.binding @wo2, set=0, binding=2, type="StorageBuffer", access="Write|Discard"
      }
    }
  }
}
```

I am guessing the `tie` is needed for the `linalg.generic` so that `@ro0` and `@wo2 get tied. It will eliminate the copy and I wont have to remove the check. Will look into it more.

Reply:
It's an interesting case. Could you show more IR dumps in previous passes? (like DispatchLinalgOnTensors and before bufferization. They can be huge; just need to extract the subset related to this.) 

> It turns out because we have the reshape into the dispatch region, and since the shape of the input and output are different, they cant be tied together in the dispatch region creation. That results in a copy here.

Hmm, I guess we can change the types of `flow.dispatch.tensor.{load|store}` and block arguments? The detailed shape actually does not matter that much from IREE's perspective; buffers are just a bag of elements. It's like we are folding `tensor_reshape` into `hal.interface.binding.subspan` ops.